test/OpenTelemetry.Tests/Logs/LogRecordSharedPoolTests.cs

```diff
@@ -183,7 +186,10 @@ public async Task ExportTest(bool warmup)
                 }
             }
 
-            using BatchLogRecordExportProcessor processor = new(new NoopExporter());
+            using var processor = new BatchLogRecordExportProcessor(new DelegatingTestExporter<LogRecord>
+            {
+                OnExportFunc = (batch) => ExportResult.Success,
```

Comment:
This is the default behavior anyway.

Reply:
You're right. I think it makes the tests easier to understand when the behavior is clearly defined. 
Plus it isolates the test from future changes in the TestExporter.
If there are strong feelings, this can be removed. :)

Reply:
> I think it makes the tests easier to understand when the behavior is clearly defined
💯

In that case we should be consistent with the usage and use the above approach `OpenTelemetryLoggerProviderTests.cs` as well.

Reply:
I intentionally left it out of that test because no methods are being called, those tests are only evaluating the ctor.
I think this is consistent with documenting the behavior of the test.

Reply:
I spoke with @utpilla offline, i'll revert the changes to the NoOpExporter and keep those as is.
This PR will just focus on the refactoring bit